### DISCUSSION TITLE
Exception in 1.3.1: Cannot invoke "java.net.URI.toString()" because "fileUri" is null

### DISCUSSION THREAD

**Discussion by frankbenoitbosch** [Genuitec/Copilot4Eclipse-community#46] [Help] [answered]
Genuitec, LLC	Copilot4Eclipse	1.3.1.202410261755	com.genuitec.copilot4eclipse.feature

java.lang.NullPointerException: Cannot invoke "java.net.URI.toString()" because "fileUri" is null
	at com.genuitec.copilot4eclipse.Copilot4Eclipse.checkCopilotFileStatus(Copilot4Eclipse.java:594)
	at com.genuitec.copilot4eclipse.CopilotEditSession.checkCopilotFileStatus(CopilotEditSession.java:307)
	at com.genuitec.copilot4eclipse.CopilotEditSession.setIsActiveEditor(CopilotEditSession.java:214)
	at com.genuitec.copilot4eclipse.CopilotEditSessionManager.partActivated(CopilotEditSessionManager.java:278)

I get this very often now, when opening editors

**Comment by wayneparrott** [comment]
Thx. We have a fix for this in the pipeline. 

Can you share which editor(s) are impacted?

W

**Reply by frankbenoitbosch** [reply]
Those that I can spot now....
adoc: Asciidoctor Editor 3.1.1 de.jcup.asciidoctoreditor.feature.group Albert Tregnaghi
xtext: Xtext UI 2.36.0.v20240825-0714 org.eclipse.xtext.ui.feature.group Eclipse Xtext
xtend: Xtend IDE 2.36.0.v20240825-0714 org.eclipse.xtend.sdk.feature.group Eclipse Xtend
lc: Launch Configuration DSL 0.7.0.202409050945 com.wamas.ide.launching.feature.feature.group SSI Schaefer IT Solutions GmbH
tpd: Target Platform Definition DSL and Generator 3.0.0.202403201545 org.eclipse.cbi.targetplatform.feature.feature.group Eclipse CBI

**Reply by wayneparrott** [reply]
Thx for sharing the editor info.

**Comment by gtjoseph** [comment]
Same issue with "Yaml Editor" and "Bash Editor"

**Reply by wayneparrott** [reply]
Thx. We are working on an update that will address this and few of the other issues reported for 1.3.1. Hope to have the update available early next week. 
W

**Comment by gekoprioritdev** [comment]
We have seen this error happening in the "SQL File Editor" as well.

**Comment by wayneparrott** [comment, marked as answer]
This NPE issue has been resolved as well as how Copilot4Eclipse handles unsupported editors more gracefully.  The NPE was limited to custom editors which Copilot4Eclipse is unable to adapt. Please update and share any issues you observe. 

**Note about Unsupported Editors**
The DTP SQL Editor is an example of such an editor. Other example editors can include multipage editors with multiple input sources, e.g., plugin.xml editor, Manifest editor... We hate excluding any editor from benefiting from Copilot assistance and will continue researching how Copilot4Eclipse can be more adaptable in future updates.

Wayne
Copilot4Eclipse Team
